src/GraphQL.php

```diff
@@ -249,23 +242,11 @@ protected function getValidationRules(): array
     /**
      * Get instance of DocumentAST.
      *
+     * @deprecated use ASTBuilder instead
      * @return \Nuwave\Lighthouse\Schema\AST\DocumentAST
      */
     public function documentAST(): DocumentAST
```

Comment:
Do people use this function outside of lighthouse?
Anyway it makes sense to mark it as deprecated first, what about removing in v5?

Reply:
Yes, depreciations will arrive in the next major by default.

Reply:
It might be used for schema manipulation/introspection - for example, in this PR we need to access the DocumentAST from BaseDirective.